crmsh/lock.py

```diff
@@ -0,0 +1,169 @@
+# Copyright (C) 2020 Xin Liang <XLiang@suse.com>
+# See COPYING for license information.
+
+
+import re
+import time
+from contextlib import contextmanager
+
+from . import utils
+from . import config
+from .msg import common_warn
+
+
+class SSHError(Exception):
+    """
+    Custom exception for ssh error
+    """
+    pass
+
+
+class ClaimLockError(Exception):
+    """
+    Custom exception if claiming lock failed or wait lock release timed out
+    """
+    pass
+
+
+class Lock(object):
+    """
+    A base class define a lock mechanism used to exclude other nodes
+    """
+
+    LOCK_DIR = "/tmp/.crmsh_lock_directory"
+    MKDIR_CMD = "mkdir {}".format(LOCK_DIR)
+    RM_CMD = "rm -rf {}".format(LOCK_DIR)
+    SSH_TIMEOUT = 10
+    SSH_OPTION = "-o ConnectTimeout={} -o StrictHostKeyChecking=no".format(SSH_TIMEOUT)
+    SSH_EXIT_ERR = 255
+
+    def __init__(self, remote_node=None, lock_func=None):
```

Comment:
I don't think we need to pass `lock_func` here. This should be part of the class, and the `RemoteLock` just would re implement the method.

Reply:
Changed